src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -512,29 +515,29 @@ protected FieldValueInfo completeValueForList(ExecutionContext executionContext,
         );
 
         List<FieldValueInfo> fieldValueInfos = new ArrayList<>();
-        int index = 0;
-        for (Object item : values) {
-            ExecutionPath indexedPath = parameters.getPath().segment(index);
-
-            ExecutionStepInfo stepInfoForListElement = executionStepInfoFactory.newExecutionStepInfoForListElement(executionStepInfo, index);
-
-            NonNullableFieldValidator nonNullableFieldValidator = new NonNullableFieldValidator(executionContext, stepInfoForListElement);
-
-            int finalIndex = index;
-            FetchedValue value = unboxPossibleDataFetcherResult(executionContext, parameters, item);
-
-            ExecutionStrategyParameters newParameters = parameters.transform(builder ->
-                    builder.executionStepInfo(stepInfoForListElement)
-                            .nonNullFieldValidator(nonNullableFieldValidator)
-                            .listSize(values.size())
-                            .localContext(value.getLocalContext())
-                            .currentListIndex(finalIndex)
-                            .path(indexedPath)
-                            .source(value.getFetchedValue())
-            );
-            fieldValueInfos.add(completeValue(executionContext, newParameters));
-            index++;
-        }
+        final SortedSet<Integer> indexes = getIndexes(getSizeIndexes(iterableValues));
+        StreamSupport.stream(iterableValues.spliterator(), true)
```

Comment:
My problem with this PR is that it introduces threading into the graphql-java.

That is `StreamSupport.stream(iterableValues.spliterator(), true)` means that it uses the common ForkJoin thread pool from platform.

As a design goal in graphql-java we have aimed to never introduce threading.  Because we cant know how this will work for saya  Reactor based app or Vertx based one or Kotlin coroutines base one and so on.

graphql-java is async (via CompleteableFutures) but it never starts a thread.  That is the job of the data fetcher call back.  it is free to dispatch data fetchers into whatever threading model you app uses.

Since `completeValueForList` is a protected method, I think it would be better for you to create your own ExecutionStrategy with this code.